packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js

```diff
@@ -159,6 +149,59 @@ export default class {
     }
   }
 
+  setClusterRegion(isRegionEnabled = false) {
+    this.cluster.region = new Datacenter({
+      name: this.cluster.region.name,
+      enabled: isRegionEnabled || this.cluster.region.enabled,
+      quota: find(this.quotas, { region: this.cluster.region.name }),
+    });
+  }
+
+  onRegionSubmit() {
+    this.setClusterRegion();
+    this.loadPrivateNetworks();
+
+    if (this.cluster.region.enabled) {
+      this.loadFlavors(this.cluster.region.name);
+    }
+
+    this.displaySelectedRegion = true;
+  }
+
+  onKubeVersionFocus() {
+    const { region } = this.cluster;
+    const { name, enabled } = region;
+    this.displaySelectedRegion = true;
+
+    if (!enabled) {
+      this.isAddingNewRegion = true;
+
+      this.Kubernetes.addRegion(this.projectId, region)
```

Comment:
Why not prefer using a `return` statement here?

Reply:
Because this function was triggered from HTML context.
We don't really need to follow how this function is end.

So the default return (undefined) is enough @antleblanc.
What do you think ?

Reply:
@antleblanc  ?

Reply:
You probably forgot @antleblanc.
As soon as you can please take a look.

Thanks.

Reply:
@mohammed-zahaf it's a good practice to return promises

Reply:
Changed, but realy i am not totaly understand benefit of this practice.